packages/slate/src/utils/deep-equal.ts

```diff
@@ -0,0 +1,37 @@
+import isPlainObject from 'is-plain-object'
+
+/*
+  Custom deep equal comparison for Slate nodes.
+
+  We don't need general purpose deep equality;
+  Slate only suports plain values and nested objects.
```

Comment:
"at the moment" is relative. It's only been this way since March, when #3532 swapped this algorithm for lodash. This PR simply reverts that change and makes the old algorithm recursive (to solve the nested object requirement).

Reply:
Adding array equals makes sense. The deep equals that this PR is replacing here does things we don't need though like comparing dates or regexp objects. It should be documented that the valid data types for these properties are json serializable things not other complex JS object types.